apps/parsley/src/context/LogContext/index.tsx

```diff
@@ -155,7 +155,9 @@ const LogContextProvider: React.FC<LogContextProviderProps> = ({
   const sectioning = useSections({
     logType: state.logMetadata?.logType,
     logs: state.logs,
-    onInitOpenSectionContainingLine: shareLine ?? state.failingLine,
+    onInitOpenSectionsContainingLines: [shareLine, state.failingLine].filter(
+      (v): v is number => v !== undefined,
```

Comment:
Wouldn't `v !== undefined` return a boolean? 
```suggestion
      (v): boolean => v !== undefined,
```

Reply:
the filter function still returns a boolean. `v is number` is a type predicate and it prevents me from having to cast the result of the filter to a number array.